### PR TITLE
Fix Gen Class Without Any Properties

### LINKED ISSUE #20: [BUG] Message without any properties is not compilable
**Describe the bug**
A proto message without any properties creates a code  that is not compilable.

**To Reproduce**

```proto
syntax = "proto3";

message Hello {
  enum Type {
    UNKNOWN = 0;
    KNOWN = 1;
  }
  message SubHello {
    message HelloExtend {
      optional Type type = 1;
    }
    optional SubHello subHello = 1;
  }
}
```

**Expected behavior**
A data class shouldn't be generated, instead it should be just `class`.

**Additional context**
Data classes require at least 1 property.

### PR DESCRIPTION
**Pull request by tamnguyenhuy**
Change fix generate class without  Any Properties

```
syntax = "proto3";

message Hello {
  enum Type {
    UNKNOWN = 0;
    KNOWN = 1;
  }
  message SubHello {
    message HelloExtend {
      optional Type type = 1;
    }
    optional SubHello subHello = 1;
  }
}

``` 

Result 

```kotlin
package testgen

import kotlinx.serialization.Serializable
import kotlinx.serialization.protobuf.ProtoNumber

@Serializable
public class Hello {
  @Serializable
  public data class SubHello(
    @ProtoNumber(number = 1)
    public var subHello: SubHello? = null,
  ) {
    @Serializable
    public data class HelloExtend(
      @ProtoNumber(number = 1)
      public var type: Type? = null,
    )
  }

  @Serializable
  public enum class Type {
    @ProtoNumber(number = 0)
    UNKNOWN,
    @ProtoNumber(number = 1)
    KNOWN,
  }
}

``` 

Fixes #20

### CODE REVIEW

**Comment by tamnguyenhuy** [comment]
@Dogacel please help to take a look at my PR and let me know your thought!!
Thanks

File: app/src/main/kotlin/dogacel/kotlinx/protobuf/gen/CodeGenerator.kt [hunk 1]
```diff
@@ -164,6 +170,19 @@ class CodeGenerator {
...
     }
 
+    /**
+     * Generate the code for the given [Descriptors.Descriptor]. Returns a Boolean determine whether the
+     * descriptor has a property field or not.
+     *
+     * @param fileDescriptor [Descriptors.FileDescriptor] to generate code for.
+     * @return Boolean that represents whether the descriptor has a property field or not.
+     */
+    private fun isHasPropertyField(fileDescriptor: Descriptors.Descriptor): Boolean {
```
Comment (Dogacel) [inline app/src/main/kotlin/dogacel/kotlinx/protobuf/gen/CodeGenerator.kt:180]: nit: 
```suggestion
    private fun hasPropertyField(messageDescriptor: Descriptors.Descriptor): Boolean {
```

File: app/src/main/kotlin/dogacel/kotlinx/protobuf/gen/CodeGenerator.kt [hunk 2]
```diff
@@ -164,6 +170,19 @@ class CodeGenerator {
...
+        return fileDescriptor.fields.any {
+            it.isOptional || it.isRequired || it.isRepeated
```
Comment (Dogacel) [inline app/src/main/kotlin/dogacel/kotlinx/protobuf/gen/CodeGenerator.kt:182]: I have a question, in proto3 if `optional int32 x = 1` is considered optional and in proto2 if `required int32 x = 1` is considered required, is this going to return false `int32 x = 1`?

I think we should have a couple of unit tests to verify this doesn't break anything.

**Review by Dogacel — COMMENTED** [review]
Happy to have you as the first contributor 🎉 

There are some minor fixes we gotta do. I am not sure if the CI/CD pipeline is ready for PRs as well. I will check it tomorrow morning.

Please let me know if you need any help, I can jump in to fix those small nits 😄

**Comment by Dogacel** [comment]
@tamnguyenhuy 

I have added some additional checks to make sure we generate the code. Also updated codecov to show coverage info.

I had to merge latest main into your PR to see those. Don't forget to fetch remote 😉

**Pull request merged**